oak_loader/src/options.rs

```diff
@@ -348,11 +348,22 @@ fn get_oidc_client_info(opt: &Opt) -> anyhow::Result<Option<ClientInfo>> {
 }
 
 /// Gets the default root TLS certificates from the embedded byte array.
-fn get_default_root_tls_certs() -> anyhow::Result<String> {
-    let result = std::str::from_utf8(include_bytes!("certs/roots.pem"))
-        .context("could not read embedded PEM-encoded root TLS certificates as a UTF8 string")?
-        .to_owned();
-    Ok(result)
+fn get_default_root_tls_certs() -> Vec<u8> {
+    include_bytes!("certs/roots.pem").to_vec()
+}
+
+/// Check that input byte vector represents a valid PEM formatted certificate.
+fn validate_certificate(certificate_bytes: &[u8]) -> anyhow::Result<()> {
```

Comment:
So why not make this function actually parse the cert as the rustls type, and pass that around instead of `Vec<u8>`? Or if that type is not good for some reason, create your own wrapper `ValidatedCert` or similar? Not sure it's worth it, but I think it would make the code more readable, since the current `Result<()>` return type is a bit weird.

see https://lexi-lambda.github.io/blog/2019/11/05/parse-don-t-validate/

Reply:
Created our own `Certificate` type. The difference between `rustls::Certificate` and `tonic::Certificate` is that the former parses a PEM file into a vector of certificates, while the latter parses a PEM file into a single Certificate object. With our internal type, we can ensure that the certificate that we pass around is a valid certificate; and can convert it into each of the destination formats when needed.